src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,82 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SPLIT_FOR_COLUMNS = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int VALUE_INDEX = 1;
+    private static final String BUY_TYPE = "buy";
+    private static final String SUPPLY_TYPE = "supply";
+    private static final String WRITE_REPORT_ERROR = "Sorry, but I can't work with this data.";
+    private static final String RESULT_TYPE = "result";
+    private static final String WRITING_TO_FILE_ERROR = "Sorry, but i can't write in file";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] dataFromFile = readFileToString(fromFileName);
+        int[] supplyBuyInfo = calculateSupplyBuy(dataFromFile);
+        String report = createReport(supplyBuyInfo);
+        writeToFile(report, toFileName);
+    }
+
+    private String[] readFileToString(String fileName) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            String line = reader.readLine();
+            StringBuilder result = new StringBuilder();
+            while (line != null) {
+                result.append(line).append(".");
```

Comment:
Why some strings for example "," separated into constants, but "." not? Also pay attenention to other stings in your solution ("\\.", "I can't read from this file any" etc)